packages/ui/src/views/repo/pull-request/compare/components/pull-request-compare-diff-list.tsx

```diff
@@ -43,31 +57,77 @@ interface PullRequestAccordionProps {
   diffMode: DiffModeEnum
   currentUser?: string
   useTranslationStore: () => TranslationStore
+  isOpen: boolean
+  onToggle: () => void
 }
 
 const PullRequestAccordion: FC<PullRequestAccordionProps> = ({
   header,
   diffMode,
   currentUser,
-  useTranslationStore
+  useTranslationStore,
+  isOpen,
+  onToggle
 }) => {
+  const { t: _ts } = useTranslationStore()
   const { highlight, wrap, fontsize } = useDiffConfig()
   const startingLine =
     parseStartingLineIfOne(header?.data ?? '') !== null ? parseStartingLineIfOne(header?.data ?? '') : null
+  const [showHiddenDiff, setShowHiddenDiff] = useState(false)
+
+  const fileDeleted = useMemo(() => header?.deleted, [header?.deleted])
+  const isDiffTooLarge = useMemo(() => {
+    if (header?.addedLines && header?.removedLines) {
+      return header?.addedLines + header?.removedLines > PULL_REQUEST_LARGE_DIFF_CHANGES_LIMIT
+    }
+    return false
+  }, [header?.addedLines, header?.removedLines])
+  const fileUnchanged = useMemo(
+    () => header?.unchangedPercentage === 100 || (header?.addedLines === 0 && header?.removedLines === 0),
+    [header?.addedLines, header?.removedLines, header?.unchangedPercentage]
+  )
+
   return (
     <StackedList.Root>
       <StackedList.Item disableHover isHeader className="cursor-default p-0 hover:bg-transparent">
-        <Accordion.Root type="multiple" className="w-full">
-          <Accordion.Item isLast value={header?.text ?? ''}>
-            <Accordion.Trigger leftChevron className="p-4 text-left">
-              <StackedList.Field title={<LineTitle text={header?.text ?? ''} />} />
-            </Accordion.Trigger>
-            <Accordion.Content>
-              <div className="flex w-full border-t">
-                <div className="w-full bg-transparent">
+        <div className="w-full border-b last:border-b-0">
+          <div
+            role="button"
+            tabIndex={0}
+            className="group flex w-full items-center justify-between p-4 text-left text-sm font-medium transition-all
+                       [&>svg]:duration-100 [&>svg]:ease-in-out"
+            onClick={onToggle}
+          >
+            <StackedList.Field title={<LineTitle text={header?.text ?? ''} isOpen={isOpen} />} />
+          </div>
+          {isOpen && (
+            <div className="border-t bg-transparent">
+              {(fileDeleted || isDiffTooLarge || fileUnchanged || header?.isBinary) && !showHiddenDiff ? (
```

Comment:
did you check that this still worked for when the file is unchanged or u have a binary file ? @shaurya-harness

Reply:
no for a PR with a binary file I was not getting this flag "isBinary" in the diff api response
couldn't replicate fileUnchanged as well